python/plugins/db_manager/dlg_sql_window.py

```diff
@@ -91,10 +91,8 @@ def __init__(self, iface, db, parent=None):
         self.updatePresetsCombobox()
 
         self.geomCombo.setEditable(True)
-        self.geomCombo.lineEdit().setReadOnly(True)
 
         self.uniqueCombo.setEditable(True)
-        self.uniqueCombo.lineEdit().setReadOnly(True)
```

Comment:
Then there is a misunderstanding:
Your change allows the user to specify __arbitrary values__ for geometry / unique id. The comboboxes are actually there to force the user to choose a valid column name, regardless of the type.

So unless there is the need for arbitrary values, e.g. because the provider can not determine the column names at all, please consider reverting these changes.